app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java

```diff
@@ -272,38 +271,38 @@ private void persistLocale(Context context, String localeCode) {
         settings.edit().putString(PREF_LOCALE, localeCode).apply();
     }
 
-    @Nullable
+    @NonNull
     public Locale getCurrentLocale(@NonNull Context context) {
         if (currentLocale != null) {
             return currentLocale;
         }
 
-        final String current = getPersistedLocale(context);
-        if (current == null) {
-            return null;
+        String current = getPersistedLocale(context);
+        if (current != null) {
+            currentLocale = Locales.parseLocaleCode(current);
```

Comment:
Thinking out loud: I'm assuming we're using Java `Locales` instead of Android resources directly so that we can support locales outside of what Android supports. However, in the short term, we don't intend to support these or an in-app locale switcher.

Should we just do away with the support for this legacy set-up? Notably, it doesn't look like Fenix uses these legacy classes either. Excepting minor cases like these, I guess it isn't bothering us so maybe it's not worth the time to remove?

Reply:
Yeah, the `Locales` file is incredibly bloated and difficult to read.  I believe there were a number of edge cases found in Fennec, but it's entirely possible that at least some of them have been fixed by the framework by now.  I would love to get rid of this, but was trying to make narrower changes, rather than refactoring everything as I went.

But yes, I would very much like to get rid of that file.